src/NuGet.Core/NuGet.ProjectModel/LockFile/AssetsLogMessage.cs

```diff
@@ -106,5 +127,25 @@ public override int GetHashCode()
 
             return combiner.CombinedHash;
         }
+
+        private class ToStringConverter<T> : JsonConverter<T>
```

Comment:
I believe you can use what's already built in: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/customize-properties?pivots=dotnet-8-0#enums-as-strings